### ISSUE TITLE
G502 loses settings with a KVM

### ISSUE THREAD

**Issue by Corionis** [libratbag/piper#908]
More of a question:  Is there a way to reset or restart?

_The problem_: Using a KVM with Linux Mint and macOS, with the "G" software, when switching back to Linux the mouse (G502) profile is lost. Have the same problem with the G910 keyboard and G13. Have a script to reset the others, but not the mouse.

Using ```systemctl restart ratbagd``` has no effect.

**Comment by staticssleever668** [comment]
Could you rephrase that?
Do you mean that device settings are not read from the mouse correctly - it shows one settings while in reality the mouse behaves differently? What if you reconnect the mouse without KVM, does the mouse still work as it's set up and does Piper show correct settings?

> Using systemctl restart ratbagd has no effect.
This makes libratbag reread device settings so that's weird.

**Comment by Corionis** [comment]
When on a Mac - that does not inherently support things like the Back/Forward buttons - then switching to Linux Mint the buttons remain disabled.

However - I just installed the Mac Mouse Fix from mousefix.org with mappings for Back/Forward and now returning to Linux does not lose those.  It seems, although I may be wrong, that switching away then back something is not detecting the event so it does not restore settings if they've been changed by the other computer.

As mentioned the Mac has the Logitech G software installed and is used.

So, because of Mouse Fix it is no longer a problem.  Sorry, I did not try bypassing the KVM as everything is wired behind my desk in the usual cable rat's nest.
